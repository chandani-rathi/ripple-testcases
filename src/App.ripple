export component App() {
	<div class='text-center'>
		<h1 class="font-semibold text-4xl">{'Welcome to Ripple!'}</h1>
		<div class='my-4 text-2xl flex items-center justify-center gap-4'>
			let $count = 0;

			<button class="hover:bg-gray-300/50 p-4 text-4xl rounded text-mid" onClick={() => $count--}>{'-'}</button>
			<span class='font-bold text-3xl w-16'>{$count}</span>
			<button class="hover:bg-gray-300/50 p-4 text-4xl rounded" onClick={() => $count++}>{'+'}</button>
		</div>
		<div>
			<p>{'This is a basic Ripple application template.'}</p>
			<p>
				{'Edit '}
				<code class="bg-gray-200/50 rounded px-1.5 py-0.5 text-gray-700 font-mono">{'src/App.ripple'}</code>
				{' to get started.'}
			</p>
		</div>
	</div>
}