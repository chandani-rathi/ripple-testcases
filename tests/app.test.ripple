import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
import { mount, flushSync, effect } from 'ripple';
import { App } from "@/App.ripple"

describe('app', () => {
	let container;

	function render(component) {
		mount(component, {
			target: container
		});
	}

	beforeEach(() => {
		container = document.createElement('div');
		document.body.appendChild(container);
	});

	afterEach(() => {
		document.body.removeChild(container);
		container = null;
	});

	it('render', () => {
		render(App);
		expect(container).toMatchSnapshot()
	});

	it('click - button 5 times', () => {
		render(App);
		const buttons = container.querySelectorAll('button');
		const count = container.querySelector("span");

        for(let i=0; i < 5; i++) {
            buttons[0].click();
            flushSync();
        }
		expect(count.textContent).toBe("-5")

	});

	it('click + button 5 times', () => {
		render(App);
		const buttons = container.querySelectorAll('button');
		const count = container.querySelector("span");

        for(let i=0; i < 5; i++) {
            buttons[1].click();
            flushSync();
        }
		expect(count.textContent).toBe("5")

	});

    

	it('click - button 3 times + button 5 times', () => {
		render(App);
		const buttons = container.querySelectorAll('button');
		const count = container.querySelector("span");

        for(let i=0; i < 3; i++) {
            buttons[0].click();
            flushSync();
        }
        for(let i=0; i < 5; i++) {
            buttons[1].click();
            flushSync();
        }
		expect(count.textContent).toBe("2")

	});
});
