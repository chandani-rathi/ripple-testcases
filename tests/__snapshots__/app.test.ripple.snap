// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`app > render 1`] = `
<div>
  <div
    class="text-center"
  >
    <h1
      class="font-semibold text-4xl"
    >
      Welcome to Ripple!
    </h1>
    <div
      class="my-4 text-2xl flex items-center justify-center gap-4"
    >
      <button
        class="hover:bg-gray-300/50 p-4 text-4xl rounded text-mid"
      >
        -
      </button>
      <span
        class="font-bold text-3xl w-16"
      >
        0
      </span>
      <button
        class="hover:bg-gray-300/50 p-4 text-4xl rounded"
      >
        +
      </button>
    </div>
    <div>
      <p>
        This is a basic Ripple application template.
      </p>
      <p>
        Edit 
        <code
          class="bg-gray-200/50 rounded px-1.5 py-0.5 text-gray-700 font-mono"
        >
          src/App.ripple
        </code>
         to get started.
      </p>
    </div>
  </div>
  
</div>
`;
